### PR TITLE
Clean train.py and adjacent files

### PR DESCRIPTION
**Pull request by bpopeters**
This PR clarifies code in train.py and adjacent files. Specifically,

1. I clarified the process in train.py's `main()`. There used to be several separate if/elses regarding `opt.train_from`. Now there's just one.
2. I snake_cased variable and function names in train.py and Loss.py.
3. I removed the `Splitter` class. It had only one function and no interesting state, so I decided that its purpose would better be implemented with a generator function. I simplified the implementation of that function, replacing some indexing logic with a fair quantity of well-commenting zipping and unzipping. The function is renamed `shards`. I'm a little bit skeptical of backpropping the variables after the last yield of a generator function, but I don't know what would be a better way to do it.
4. Where possible, I removed `opt` from the argument signatures of functions. It's rarely necessary to pass all the script's arguments.
5. I changed the implementation of `aeq`.

### CODE REVIEW

**Comment by bpopeters** [comment]
6. Add a `use_gpu` function to Utils.py that takes `opt` as an argument and returns whether to use the gpu.

**Comment by srush** [comment]
Looks great, thanks as always!

- Would be nice in the future to have it print out parameter numbers at an even finer granularity. 
- Also I think more loss functions are coming, so we might eventually want to modularize that part of the code as well. It's nice that it doesn't get saved.
